### PR TITLE
Academic Mentorship mentor-mentee mappings

### LINKED ISSUE #141: Academic Mentorship mentor-mentee mappings
## Decision Summary

- Domain name: `Academic Mentorship`; School page tab label remains `Mentorship`.
- Permission key: replace old `mentorship` with `academic_mentorship`.
- Access model: Admin and Program Admin manage mappings; read-only Program Admins view only; Teachers and PMs use only the School page view; passcode users see no Mentorship tab.
- Program gating: use a dedicated Academic Mentorship allowlist constant, not `NVS_GATED_FEATURES`; v1 value is `[*]` for all Programs.
- Admin route: `/admin/academic-mentorship`; Program Admins may enter `/admin` only to discover this card.
- API namespace: `/api/academic-mentorship/*` with one shared server-side access helper for role, School scope, Program allowlist, `read_only`, and action checks.
- Storage table: `academic_mentorship_mentor_mentee_mappings`, added by db-service migration; no db-service REST endpoint is needed.
- LMS write path: LMS route handlers write the mapping table directly through the shared Postgres helper because the table has no db-service REST API.
- Mentor id: store `mentor_user_id` referencing `user.id`; join Staff Management Teacher data only for eligibility and display.
- Mentee id: store internal `student.id`; CSV/search/display may use external `student.student_id`.
- Active uniqueness: one active Mapping per Student per School and academic year, enforced by a partial unique index where `ended_at IS NULL`.
- History: remove and reassign end the active row and preserve historical rows; timestamps are system action time with no backdating.
- Reassign: dedicated atomic transaction ending the old Mapping and creating the new Mapping.
- CSV upload: template headers are `mentor_email,student_id`; upload uses the selected academic year; validation is all-or-nothing and capped at 2,000 data rows.
- Staff Management blockers: permission/user deletion blocks any Teacher with Mapping history; Teacher exit/revoke blocks only active Mentees.

## Problem Statement

Avanti Fellows needs Academic Mentorship in LMS so selected Students can be assigned to Academic Mentors by School and academic year. The current School page has only a placeholder Mentorship tab, the existing permission key is a generic `mentorship`, and there is no durable Mentor-Mentee Mapping workflow with history, CSV upload, or Staff Management safeguards.

Without this workflow, Admins and Program Admins cannot centrally maintain Academic Mentor assignments, Teachers cannot see their current Mentees in LMS, PMs cannot inspect current mappings during school operations, and Staff Management can accidentally remove or exit Teachers who still carry mentorship responsibility.

## Solution

Build an LMS-owned Academic Mentorship workflow backed by a new db-service migration. Admins and eligible Program Admins manage Mentor-Mentee Mappings from `/admin/academic-mentorship`; Teachers and PMs consume read-only current-year views on the School page Mentorship tab. Mapping actions preserve history, enforce one active Mentor per Student per School/year, support CSV upload for bulk assignment, and block Staff Management operations that would orphan active or historical Academic Mentor-Mentee Mappings.

The feature must use project vocabulary and existing LMS patterns: direct Postgres reads through the shared query helper, direct Postgres writes for this LMS-owned workflow table, thin route handlers, centralized permission checks, canonical School roster rules for Mentee eligibility, and behavior-focused Vitest coverage.

## User Stories

1. As an Admin, I want to assign Academic Mentors to selected Students for a School and academic year, so that mentorship ownership is recorded in LMS.
2. As an Admin, I want to view active mappings grouped by Academic Mentor, so that I can quickly inspect each Mentor's Mentee count.
3. As an Admin, I want to include historical mappings on demand, so that I can audit prior assignments without cluttering the default view.
4. As an Admin, I want to reassign a Mentee from one Academic Mentor to another in one action, so that the old Mapping ends and the new Mapping begins consistently.
5. As an Admin, I want to remove an active Mapping with confirmation, so that a Student can intentionally have no active Academic Mentor.
6. As an Admin, I want to bulk upload mappings by `mentor_email` and external `student_id`, so that large assignment lists can be created without repetitive manual entry.
7. As an Admin, I want upload validation to report every row error at once, so that I can fix the CSV in one pass.
8. As an Admin, I want failed uploads to provide an error CSV containing rejected rows and reasons, so that I can correct the source file.
9. As a Program Admin, I want to manage mappings only within my School and Program scope, so that I can support my assigned Schools without broader access.
10. As a read-only Program Admin, I want to view mappings without edit controls, so that my `read_only` access is enforced in both UI and API.
11. As a Program Admin, I want `/admin` to show only the Academic Mentorship management entry available to me, so that existing admin-only tools remain restricted.
12. As a Teacher, I want to see only my current active Mentees on the School page Mentorship tab, so that I know which Students I am responsible for this academic year.
13. As a Teacher, I want an empty state when I have no Mentees, so that I know there is no current assignment rather than a loading or access issue.
14. As a PM, I want read-only current active mappings on the School page, so that I can understand mentorship coverage during school operations.
15. As an Admin or Program Admin, I want a Manage mappings link from the School page that preselects the School and current academic year, so that I can move from overview to management without re-filtering.
16. As Staff Management, I want to block deleting a Teacher with any Academic Mentor-Mentee Mapping history, so that audit history does not point at a removed person.
17. As Staff Management, I want to block exit/revoke only when a Teacher has active Mentees, so that historical mappings can remain after the Teacher exits.
18. As an implementer, I want one shared Academic Mentorship access helper, so that every route enforces role, School scope, Program allowlist, and read-only behavior consistently.

## Implementation Decisions

- Permission model: add `academic_mentorship` to the feature permission matrix and migrate School page/admin checks away from `mentorship`. Teachers and PMs get view access only; Admins and Program Admins get edit unless `read_only` downgrades them. Passcode users get no access.
- Program gating: add an Academic Mentorship Program allowlist constant with wildcard `[*]` for v1. Do not add this feature to the existing NVS/CoE/Nodal gated feature set. The data model and queries must work for all Programs, including NVS and PMU Schools.
- Access helper: centralize checks in a server-only Academic Mentorship helper that resolves the signed-in user, selected School, School scope, Program allowlist, role, `read_only`, and requested action. Route handlers should call this helper before any data access.
- School page: replace the placeholder Mentorship content with role-specific Academic Mentorship views. Teachers see their own active current-year Mentees sorted by grade then name. PMs see current active mappings read-only. Admins and Program Admins see a read-only overview plus a Manage mappings link.
- Admin landing page: allow Program Admins to enter `/admin` when they can view Academic Mentorship, but show only the Academic Mentorship card unless the user is a real Admin. Existing admin-only cards remain Admin-only.
- Admin management page: `/admin/academic-mentorship` has School and academic year pickers. School auto-selects when exactly one accessible School exists. Academic year defaults to the current LMS academic year and offers current plus two prior years. Unsupported URL years are viewable only if needed but edits are blocked.
- Query params: selected School and academic year are reflected in query params such as `school_code` and `academic_year`; School page Manage mappings links prefill both.
- Mapping list: default to active Mappings only, grouped by Academic Mentor. A Show history toggle includes historical rows in the same grouped view. Only Mentors with rows in the selected School/year/view are shown.
- Row display: show Mentee name, grade, external `student.student_id`, assigned date, and active/historical status. When history is enabled, also show ended date. Do not display `program_id`, `assigned_by_user_id`, or `ended_by_user_id` by default in v1.
- Manual create: Add Mapping shows only active, unassigned Mentees from the selected School/year and eligible Academic Mentors from the same selected School. It never implicitly reassigns an already mapped Student.
- Manual concurrency: if create hits the active uniqueness constraint, return `Student already has a mentor mapped` and refresh the table.
- Reassign: expose Reassign only on active rows for users with edit access. The replacement Mentor selector excludes the current Mentor. The action must run in one transaction: end the old Mapping and insert the new Mapping.
- Remove: expose Remove only on active rows for users with edit access. Require confirmation that the Student will no longer have an active Academic Mentor. Removal sets `ended_at` and `ended_by_user_id`; v1 UI does not ask for `end_reason`.
- Mentor eligibility: eligible Academic Mentors are completed Staff Management Teachers only: active LMS permission with Teacher role, real AF `teacher` row, non-exited Staff Management state, and effective access to the selected School. Mentor search is by name and email. Existing PM teacher endpoints are not sufficient because they do not join completed Staff Management state.
- Mentee eligibility: use the canonical School roster/current-academic-year rules, excluding dropout Students and Students who already have an active Mapping for the selected School/year. Mentee search is by name and external `student.student_id`.
- Mentor and Mentee Program relation: both must be valid within the selected School and allowed Academic Mentorship Program scope, but they do not need to be in the same Program. New Mappings store the Mentee roster Program when available; otherwise `program_id` is null.
- Schema: db-service migration creates `academic_mentorship_mentor_mentee_mappings` with `id`, `school_id`, nullable `program_id`, `academic_year`, `mentor_user_id`, `student_id`, `assigned_at`, `assigned_by_user_id`, `ended_at`, `ended_by_user_id`, optional `end_reason`, `inserted_at`, and `updated_at`.
- Foreign keys: `school_id`, nullable `program_id`, `mentor_user_id`, `student_id`, `assigned_by_user_id`, and `ended_by_user_id` reference the existing shared tables. Actor columns reference `user.id`, not `user_permission.id`.
- Indexes: add the partial unique active-Mapping index on `(school_id, academic_year, student_id)` where `ended_at IS NULL`; add read-path indexes for School/year active Mappings, Mentor/year active Mappings, and Teacher history checks.
- API contract: route handlers under `/api/academic-mentorship/*` should cover page bootstrap, mapping list, Mentor options, Mentee options, manual create, remove, reassign, template download, and CSV upload. Mutations return structured success/error responses that the UI can use for messages and refreshes.
- CSV template: download a CSV with headers `mentor_email,student_id`.
- CSV parsing: require `mentor_email` and `student_id`; allow and preserve extra columns for error CSV output; ignore completely blank rows; trim both required fields; match Mentor email case-insensitively; match external Student id by trimmed exact text.
- CSV validation: empty files or missing required headers return file-level errors, not an error CSV. Rows with one required field blank get row-level errors. Duplicate `student_id` values in the CSV fail the whole upload and identify duplicate rows using spreadsheet row numbers.
- CSV transaction: validate all rows before writing. If any row is invalid or any target Student already has an active Mentor, write nothing. Successful upload inserts all rows in one transaction, returns a success count, and refreshes the mapping table.
- CSV error output: failed row-level uploads allow downloading an error CSV containing only rejected rows, the original uploaded columns, and an `error_reason` column.
- Staff Management delete/revoke: before deleting a user permission or otherwise hard-removing a Teacher from Staff Management, check for any Mapping history and block with a message. Before Teacher exit/revoke, block only if active Mentees exist. Warning/blocking messages include active Mentee count and a management link for the relevant School/year when available.
- Data access: all SQL uses placeholders and the shared query/transaction helpers. Student, batch, quiz-session, and document writes remain out of scope and still go through db-service as today.
- ADR check: no conflict found. Existing ADRs support db-service owning shared migrations while LMS owns authenticated, permission-aware APIs.

## Testing Decisions

- Test behavior and contracts, not implementation details: verify visible access, API status codes, validation errors, transaction outcomes, and returned data shapes.
- Permission tests: cover `academic_mentorship` matrix behavior for Admin, Program Admin, read-only Program Admin, PM, Teacher, passcode user, School scope, and wildcard Program allowlist. Existing permission tests are the prior style.
- Access helper tests: cover allowed/denied combinations for view/edit actions, unsupported School/year edits, and read-only downgrade before data access.
- Mapping domain tests: cover active uniqueness handling, manual create, remove, reassignment transaction, historical preservation, actor ids, and `program_id` auto-fill/null fallback. Use the Staff Management transaction tests as prior art.
- Mentor eligibility tests: cover completed Staff Management Teacher requirements, exited/revoked Teachers, Teachers without `teacher` rows, School scope through centre seats, and searchable name/email output.
- Mentee eligibility tests: cover canonical current-year roster filtering, dropout exclusion, already-mapped exclusion, Student id search, and sorting by grade/name.
- CSV tests: cover required headers, empty file, blank rows, partial blank rows, trimming, case-insensitive Mentor email, exact Student id matching, duplicate Student ids, 2,000-row cap, all-errors-at-once validation, all-or-nothing writes, and error CSV shape. Centre import and curriculum CSV export tests are prior art.
- API route tests: follow existing route-handler Vitest style with mocked session/access helpers. Cover 401/403, invalid JSON/body, invalid query params, success responses, conflict responses, and no data access before authorization.
- UI/component tests: cover admin landing visibility for Admin vs Program Admin, admin management view/edit modes, School page role-specific Mentorship tab content, Manage mappings prefilled link, confirmations, success messages, and automatic refresh calls.
- Staff Management tests: add delete/history and exit/active-Mentee blockers to existing Staff Management and user deletion test coverage.
- No broad E2E suite is required for v1 unless implementation risk grows; focused unit/component/API tests are enough for the first slice.

## Out of Scope

- Narrowing exact Academic Mentorship Program ids beyond the v1 `[*]` allowlist.
- A db-service REST endpoint for Academic Mentorship Mappings.
- AI app schema or AI Mentorship Guide changes.
- General Export mappings CSV action.
- Backdating assignment or removal timestamps.
- End reason entry in the v1 UI.
- Mapping history on the School page for Teachers or PMs.
- Showing `program_id`, `assigned_by_user_id`, or `ended_by_user_id` in default admin rows.
- Uploads above 2,000 data rows.
- Reusing global CoE/Nodal/NVS gating for this feature.

## Further Notes

- Keep Academic Mentorship distinct from existing `acad_mentorship_*` report-generation tables while using the broader Academic Mentorship product language.
- Known future Program lookup: if the wildcard is narrowed later, likely include all `TP-Sync` Programs except Enable Plus, and Juned PMU may be all `TP-Async` except JNV NVS and TN-related Programs. This requires confirmation and is not blocking v1.
- `CLAUDE.md` is a symlink to `AGENTS.md`; project non-negotiables apply: permission gate before data access, parameterized SQL only, and no server-only imports in client components.

### LINKED ISSUE #145: Academic Mentorship: show guarded admin mapping overview
AFK: true
Parent: #141
Council review: skipped (reviewRounds=0)

## Slice summary

Build the first usable Academic Mentorship LMS path: eligible Admin and Program Admin users can enter `/admin/academic-mentorship`, choose a School and academic year, and view grouped Academic Mentor-Mentee Mappings with an empty state when none exist.

This slice owns the LMS permission/access foundation and the read-only admin overview. The database table contract is handled separately in #151.

## Blocked by

#151

## Acceptance criteria

- [ ] Add `academic_mentorship` as the feature key and stop using the generic `mentorship` key for this workflow.
- [ ] Academic Mentorship uses its own Program allowlist constant with v1 wildcard behavior for all Programs, and is not included in the existing NVS-gated feature set.
- [ ] Passcode users have no Academic Mentorship access; Admin and Program Admin can view/edit unless `read_only` downgrades edit to view; Teachers and PMs are view-only for later School-page use.
- [ ] A shared server-only Academic Mentorship access helper gates view/edit actions by session, role, School scope, Program allowlist, and `read_only` before any mapping data access.
- [ ] `/admin` allows Program Admins who can view Academic Mentorship to enter and shows only the Academic Mentorship card for them; real Admins keep the existing admin cards plus Academic Mentorship.
- [ ] `/admin/academic-mentorship` supports School and academic year selection, reflects `school_code` and `academic_year` in the URL, auto-selects the only accessible School, and offers current plus two prior academic years.
- [ ] The overview defaults to active mappings only, groups rows by Academic Mentor, shows each group Mentee count, and shows Mentee name, grade, external `student_id`, assigned date, and status.
- [ ] A Show history toggle includes historical mappings in the same grouped view and shows ended date for historical rows.
- [ ] Unsupported academic years do not allow edit mode.
- [ ] The LMS read path expects `academic_mentorship_mentor_mentee_mappings` from #151 and does not add a db-service REST endpoint.

## Testing guidance

- Add focused tests for `getFeatureAccess`/permission matrix behavior, wildcard Program allowlist behavior, passcode denial, and `read_only` downgrade.
- Add access helper tests proving denied requests do not query mapping data.
- Add route tests for 401, 403, invalid School/year params, default active-only listing, Show history listing, and empty state response.
- Add Admin page tests for Admin vs Program Admin card visibility.
- Add management page/component tests for School/year picker behavior, URL params, grouped rows, Mentee count, Show history, and view-only mode.

## Out of scope

- db-service migration work, covered by #151.
- Manual add, remove, and reassign actions.
- CSV template or upload.
- School page Mentorship Tab content.
- Staff Management delete/exit blockers.
- General Export mappings CSV.

### LINKED ISSUE #146: Academic Mentorship: add and remove mappings manually
AFK: true
Parent: #141
Council review: skipped (reviewRounds=0)

## Slice summary

Allow Admin and editable Program Admin users to create and end Academic Mentor-Mentee Mappings manually from `/admin/academic-mentorship`, with eligibility pickers and immediate table refresh.

## Blocked by

- Blocked by #145

## Acceptance criteria

- [ ] Editable users see Add Mapping controls on the admin management page; read-only users can view mappings but see no mutation controls.
- [ ] The Academic Mentor selector lists only eligible completed Staff Management Teachers for the selected School: active teacher permission, real AF `teacher` record, non-exited Staff Management state, and effective access to the selected School.
- [ ] The Academic Mentor selector is searchable by mentor name and email.
- [ ] The Mentee selector lists active, unassigned Students from the selected School and academic year using the canonical School roster rules, excluding dropouts and Students with an active Mapping.
- [ ] The Mentee selector is searchable by Student name and external `student.student_id`.
- [ ] Creating a Mapping stores `school_id`, selected `academic_year`, `mentor_user_id`, internal `student.id`, system `assigned_at`, `assigned_by_user_id`, and nullable `program_id` from the Mentee roster Program when available.
- [ ] Manual add never implicitly reassigns an already mapped Student.
- [ ] If a concurrent create hits the active uniqueness constraint, the API returns `Student already has a mentor mapped` and the UI refreshes the table.
- [ ] Removing an active Mapping requires confirmation that the Student will no longer have an active Academic Mentor, then sets `ended_at` and `ended_by_user_id`.
- [ ] Historical mapping rows show no Remove action.
- [ ] Successful add/remove actions show a small success message and refresh the mapping table automatically.

## Testing guidance

- Add domain/helper tests for mentor eligibility, mentee eligibility, `program_id` autofill/null fallback, active uniqueness conflict handling, and historical preservation.
- Add route tests for mentor options, mentee options, create, remove, 401/403, invalid payloads, read-only denial, and no data access before authorization.
- Add component tests for searchable selectors, hidden controls in view-only mode, confirmation copy, conflict message, success message, and refresh calls.

## Out of scope

- Reassignment.
- CSV template or upload.
- School page Mentorship Tab content.
- Staff Management delete/exit blockers.
- Backdating assignment or removal timestamps.
- End reason entry in the v1 UI.

### LINKED ISSUE #147: Academic Mentorship: reassign active mentees
AFK: true
Parent: #141
Council review: skipped (reviewRounds=0)

## Slice summary

Add the dedicated Reassign action for active Academic Mentor-Mentee Mappings so an editable Admin or Program Admin can move one Mentee from the current Academic Mentor to another in a single atomic action.

## Blocked by

- Blocked by #146

## Acceptance criteria

- [ ] Active mapping rows show Reassign only for users with edit access.
- [ ] Historical mapping rows show no Reassign action.
- [ ] The replacement Academic Mentor selector includes eligible Academic Mentors for the selected School and excludes the current Academic Mentor.
- [ ] Reassign requires confirmation that the old Mapping will end and a new Mapping will begin.
- [ ] The API performs reassignment in one transaction: set `ended_at` and `ended_by_user_id` on the old active Mapping, then insert the new Mapping with system `assigned_at`, `assigned_by_user_id`, and Mentee roster `program_id` when available.
- [ ] The old Mapping remains visible only when Show history is enabled; the new Mapping appears in the active default view.
- [ ] If the Student is already mapped by a concurrent action, the UI shows `Student already has a mentor mapped` and refreshes the table.
- [ ] Read-only users and out-of-scope users cannot reassign via UI or direct API call.

## Testing guidance

- Add transaction/domain tests that prove the old Mapping is ended and the new Mapping is inserted together, with rollback on insert failure.
- Add route tests for success, missing/invalid replacement mentor, current mentor exclusion, inactive/historical mapping denial, conflict response, 401/403, and read-only denial.
- Add component tests for Reassign visibility, confirmation, selector exclusion, success message, conflict message, and table refresh.

## Out of scope

- Manual add/remove behavior beyond the shared helpers needed by reassignment.
- CSV upload.
- Backdating timestamps.
- End reason entry in the v1 UI.

### LINKED ISSUE #148: Academic Mentorship: import mappings by CSV
AFK: true
Parent: #141
Council review: skipped (reviewRounds=0)

## Slice summary

Add the Academic Mentorship CSV template and upload flow so editable Admin and Program Admin users can bulk-create Mentor-Mentee Mappings for the selected School and academic year.

## Blocked by

- Blocked by #146

## Acceptance criteria

- [ ] The admin management page provides a CSV template download with `mentor_email,student_id` headers.
- [ ] CSV upload uses the page-selected School and academic year; CSV files do not include an academic year column.
- [ ] Upload requires `mentor_email` and `student_id` headers, allows extra columns, and preserves extra columns in error CSV output.
- [ ] Empty files or missing required headers return a normal file-level error, not an error CSV.
- [ ] Completely blank rows are ignored.
- [ ] Rows with one required field blank produce row-level validation errors.
- [ ] `mentor_email` and `student_id` are trimmed; mentor email lookup is case-insensitive; external `student.student_id` matching is trimmed exact text.
- [ ] Upload is capped at 2,000 data rows.
- [ ] Duplicate `student_id` values in the file fail the upload and report duplicate row numbers using spreadsheet row numbers.
- [ ] Each row validates that `mentor_email` belongs to an eligible Academic Mentor for the selected School and that `student_id` belongs to an active eligible unassigned Mentee for the selected School/year.
- [ ] If any row is invalid or any target Student already has an active Mapping, no rows are written.
- [ ] A successful upload inserts all rows in one transaction, shows a success count, and refreshes the mapping table automatically.
- [ ] Row-level failures allow downloading an error CSV containing only rejected rows, the original uploaded columns, and `error_reason`.
- [ ] Read-only users and out-of-scope users cannot upload via UI or direct API call.

## Testing guidance

- Add parser/validation tests for required headers, empty file, blank rows, partial blanks, trimming, case-insensitive mentor email, exact Student id matching, duplicate Students, 2,000-row cap, row numbers, and error CSV shape.
- Add transaction tests for all-or-nothing writes and active Mapping conflict handling.
- Add route tests for template download, upload success, file-level errors, row-level errors, 401/403, read-only denial, and no data access before authorization.
- Add component tests for template button, upload error display, error CSV download, success count, and table refresh.

## Out of scope

- General Export mappings CSV.
- Manual add/remove UI changes except reuse of existing helpers.
- Reassignment.
- Uploads above 2,000 data rows.
- Backdating assignment timestamps.

### LINKED ISSUE #149: Academic Mentorship: show current mappings on the School page
AFK: true
Parent: #141
Council review: skipped (reviewRounds=0)

## Slice summary

Replace the School page Mentorship placeholder with role-specific Academic Mentorship views for the current academic year.

## Blocked by

- Blocked by #145

## Acceptance criteria

- [ ] The School page keeps the tab label `Mentorship`, but the feature key and domain language are Academic Mentorship.
- [ ] Passcode users do not see the Mentorship Tab.
- [ ] Teachers see only their own current active Mentees for the current academic year as a flat list sorted by grade, then name.
- [ ] Teachers with no current active Mentees see `No mentees assigned for this academic year.`
- [ ] Program Managers see active Academic Mentor-Mentee Mappings for the School and current academic year in read-only mode.
- [ ] Admins and Program Admins see a read-only current-year overview on the School page.
- [ ] Admins and Program Admins see a Manage mappings link to `/admin/academic-mentorship` with `school_code` and the current academic year prefilled.
- [ ] Read-only Program Admins can follow the Manage mappings link but land in view-only mode.
- [ ] Program Managers do not see the Manage mappings link.
- [ ] The School page does not show mapping history in v1.
- [ ] Client components used for the tab do not import server-only modules such as `@/lib/db` or `@/lib/permissions`.

## Testing guidance

- Add School page tests for Teacher populated list, Teacher empty state, PM read-only overview, Admin/Program Admin overview, Manage mappings link params, read-only Program Admin behavior, PM no-link behavior, and passcode tab hiding.
- Add API/helper tests for current-year filtering, Teacher-only active Mentees, School scope, Program allowlist behavior, 401/403, and no data access before authorization.
- Add component tests for sorting by grade/name and hidden history.

## Out of scope

- Admin management mutations.
- Mapping history on the School page.
- General export.
- AI app or AI Mentorship Guide changes.

### LINKED ISSUE #150: Academic Mentorship: block Staff Management deletes and exits
AFK: true
Parent: #141
Council review: skipped (reviewRounds=0)

## Slice summary

Protect Academic Mentor-Mentee Mapping history from Staff Management operations that would delete or exit Teachers with mentorship responsibility.

## Blocked by

- Blocked by #145

## Acceptance criteria

- [ ] Deleting a Teacher-related `user_permission` or otherwise hard-removing a Teacher from Staff Management is blocked when that Teacher has any Academic Mentor-Mentee Mapping history.
- [ ] Teacher exit/revoke is blocked only when the Teacher has active Mentees.
- [ ] Teacher exit/revoke remains allowed when the Teacher has only historical ended Mappings.
- [ ] Blocking responses include the active Mentee count when active Mentees are the reason.
- [ ] Blocking messages include a link to `/admin/academic-mentorship` for the relevant School and academic year when available.
- [ ] The checks use `mentor_user_id`/`user.id`, not `user_permission.id`, to match the Mapping audit model.
- [ ] Existing Staff Management last-seat and seated-user protections still behave as before.
- [ ] Out-of-scope users still receive the existing 401/403 responses before any mentorship blocker queries run.

## Testing guidance

- Add or extend `src/lib/staff-admin.test.ts` coverage for exit/revoke blocked by active Mentees, allowed with only historical Mappings, message content, count, and management link.
- Add or extend `src/app/api/admin/users/[id]/route.test.ts` coverage for user deletion blocked by any Mapping history and no blocker query before admin authorization.
- Add tests that actor columns and blocker lookups use `user.id`/`mentor_user_id`, not `user_permission.id`.
- Add regression coverage for existing last-seat and seated-user behavior if touched.

## Out of scope

- Creating, removing, or reassigning Academic Mentor-Mentee Mappings.
- Changing Staff Management roster layout beyond the blocking/warning text.
- Deleting mapping history.

### LINKED ISSUE #151: Academic Mentorship: add mentor-mentee mapping schema
AFK: true
Parent: #141
Council review: skipped (reviewRounds=0)

## Slice summary

Add the db-service schema for Academic Mentor-Mentee Mappings. This slice is intentionally db-service-only: create the database migration for `academic_mentorship_mentor_mentee_mappings` in the sibling db-service repo/worktree, and do not add an LMS REST endpoint for this table.

Ralph is tracking this slice from the LMS parent issue, but the implementation work belongs in db-service. Use `/Users/deepanshmathur/Documents/AF/db-service_wktree_main` for the code changes unless the user points to a different db-service worktree.

## Blocked by

None - can start immediately.

## Acceptance criteria

- [ ] Add a db-service migration creating `academic_mentorship_mentor_mentee_mappings`.
- [ ] Include columns: `id`, `school_id`, nullable `program_id`, `academic_year`, `mentor_user_id`, `student_id`, `assigned_at`, `assigned_by_user_id`, `ended_at`, `ended_by_user_id`, optional `end_reason`, `inserted_at`, and `updated_at`.
- [ ] Add FK references for `school_id`, nullable `program_id`, `mentor_user_id`, `student_id`, `assigned_by_user_id`, and `ended_by_user_id`.
- [ ] `assigned_by_user_id` and `ended_by_user_id` reference `user.id`, not `user_permission.id`.
- [ ] Add partial unique index on `(school_id, academic_year, student_id)` where `ended_at IS NULL`.
- [ ] Add read-path indexes for School/year active mappings, Mentor/year active mappings, and Teacher history checks.
- [ ] Do not add a db-service REST endpoint.
- [ ] Do not change the AI app.

## Testing guidance

- Add the smallest migration/schema verification used by db-service for migrations.
- Verify the partial unique index and FK definitions are present.
- Run the focused db-service migration/test command for this migration.

## Out of scope

- LMS permission key, APIs, UI, CSV upload, and Staff Management blockers.
- Any db-service REST route or controller.
- Any AI app schema/code change.

### PR DESCRIPTION
**Pull request by deepansh96**
## Summary

- Adds the Academic Mentorship feature key, allowlist, shared access helper, and guarded `/admin/academic-mentorship` management page.
- Adds grouped active/history Mentor-Mentee mapping views plus manual add, remove, and reassign flows with server-side eligibility checks.
- Adds CSV template/download and all-or-nothing CSV import for mappings by `mentor_email` and external `student_id`.
- Replaces the School page Mentorship placeholder with role-specific current-year Academic Mentorship views and a prefilled management link for eligible Admin/Program Admin users.
- Blocks Staff Management delete/exit paths that would orphan Academic Mentor-Mentee mapping history or active mentees.
- Adds focused route, component, permission, staff-admin, and domain tests for the new workflow.

## Linked Issues

- Closes #141
- Closes #145
- Closes #146
- Closes #147
- Closes #148
- Closes #149
- Closes #150
- Closes #151

## Final Review

- Outcome: pass.
- Quality checks from final review passed: `npm test`, `npm run lint`, and `npm run build`.
- Acceptance criteria for #145, #146, #147, #148, #149, #150, and sibling db-service schema issue #151 were verified as satisfied.
- No blocking findings were found.
- Non-blocking risk noted: duplicate current-grade `enrollment_record` rows in source data could duplicate mapping display/option rows because `src/lib/academic-mentorship.ts` joins current-grade rows directly rather than using the roster helper's dedupe behavior.

## Human QA Checklist

- [ ] Log in as Admin and confirm `/admin` shows the Academic Mentorship card alongside existing admin tools.
- [ ] Log in as Program Admin and confirm `/admin` shows only the Academic Mentorship entry, with edit controls hidden when `read_only` is set.
- [ ] On `/admin/academic-mentorship`, choose a School/year and verify active mappings are grouped by Academic Mentor with Mentee counts.
- [ ] Toggle Show history and confirm ended mappings appear without Add/Remove/Reassign actions.
- [ ] Add a mapping with eligible mentor and unassigned mentee, then confirm the table refreshes and the mentee disappears from the unassigned selector.
- [ ] Remove an active mapping and confirm it moves to history only after Show history is enabled.
- [ ] Reassign an active mentee and confirm the old mentor is excluded from replacement choices and the new active mapping is shown.
- [ ] Download the CSV template, upload a valid `mentor_email,student_id` file, and confirm the success count plus refreshed table.
- [ ] Upload an invalid CSV with duplicate or unknown students and confirm no rows are written and an error CSV is available.
- [ ] Open a School page as Teacher, PM, Admin, Program Admin, and passcode user to verify role-specific Mentorship tab visibility/content.
- [ ] Try deleting/exiting a Teacher with active and historical mappings from Staff Management and confirm the expected block/allow behavior.

### CODE REVIEW

**Comment by deepansh96** [comment]
@codex review

File: src/app/api/academic-mentorship/mappings/import/route.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  try {
+    formData = await request.formData();
+  } catch {
+    return academicMentorshipError("Invalid multipart form data", 400);
+  }
+
+  const parsed = parseSchoolYear({
+    schoolCode: formString(formData, "school_code"),
+    academicYear: formString(formData, "academic_year"),
+    requireEditable: false,
```
[the remark this answers is not in the document]
Comment (deepansh96) [reply src/app/api/academic-mentorship/mappings/import/route.ts:58]: Fixed in 569cd91: CSV import now requires the supported Academic Mentorship year list before access/import, with tests covering unsupported year rejection.

File: src/lib/academic-mentorship.ts [hunk 2]
```diff
@@ -0,0 +1,1221 @@
...
+      const insertParams = academicMentorshipImportParams(
+        params,
+        rows,
+        mentorByEmail,
+        menteeByStudentId
+      );
+      const inserted = await client.query<{ id: number | string }>(
+        `INSERT INTO academic_mentorship_mentor_mentee_mappings
+           (school_id, academic_year, mentor_user_id, student_id, program_id, assigned_by_user_id, assigned_at, inserted_at, updated_at)
+         VALUES ${values}
```
[the remark this answers is not in the document]
Comment (deepansh96) [reply src/lib/academic-mentorship.ts:933]: Fixed in 569cd91: non-current CSV imports now set ended_at and ended_by_user_id, so prior-year backfills are historical and Staff Management does not treat them as active mentees.

File: src/components/PerformanceTab.tsx [hunk 3]
```diff
@@ -302,6 +333,25 @@ export default function PerformanceTab({ schoolUdise }: Props) {
               </option>
             ))}
           </Select>
+
+          {/* Test Grade filter — the grade the test targets, which can differ
+              from the students' grade (e.g. a grade-12 batch sitting an
+              11th-grade test). Options come from the loaded test set. */}
+          {selectedGrade != null && !deepDiveSession && availableTestGrades.length > 0 && (
```
[the remark this answers is not in the document]
Comment (deepansh96) [reply src/components/PerformanceTab.tsx:340]: Not related to this Academic Mentorship PR. This is a performance cumulative-view URL filter issue from the existing/base code, so I am leaving it for a separate PR.

File: src/components/performance/StudentResultsTable.tsx [hunk 4]
```diff
@@ -1,17 +1,139 @@
...
+  studentQuestions: StudentQuestionRow[]
+): StudentQuestionRow[] {
+  const byId = ch.chapter_id
+    ? studentQuestions.filter((q) => q.chapter_id === ch.chapter_id)
+    : [];
+  if (byId.length > 0) return byId;
+  const chName = ch.chapter_name.toLowerCase();
+  return studentQuestions.filter((q) => {
+    const qName = q.chapter_name.toLowerCase();
+    return qName === chName || chName.endsWith(qName) || qName.endsWith(chName);
```
[the remark this answers is not in the document]
Comment (deepansh96) [reply src/components/performance/StudentResultsTable.tsx:50]: Not related to this Academic Mentorship PR. This is a performance question drill-down subject-matching issue from the existing/base code, so I am leaving it for a separate PR.
